### ISSUE TITLE
SVG Image Rendering/ Refresh too slow e.g. vs. Web browser

### ISSUE THREAD

**Issue by EugeneSindambiwe** [fyne-io/fyne#3114]
### Describe the bug:
A simple SVG image generated under 5 ms can't be refreshed within under 60 s in a Fyne app. A larger, but alo simple image (from rendering perspective) can't even come up in a Fyne app. In both cases  we used standard browsers as benchmark. They do the job very easily.

### How To Reproduce:
Steps to reproduce the behaviour:
1. Simple image with e.g. 20 small squares in different colors (we used 20 in gray, red, yellow, green or red) with a few lines connecting some of the squares.
2. We built color randomized so we can easily notice image change at refresh
3. A simple Fyne App with that image as canvas contains also a go routine to refresh the image at each tick. A Refresh() is called only for that image.
4. The image generation takes 4--5 ms
5. The best image refresh frequency we could get was 50-60 sec.
6. The larger, but simple SVG image is a bar diagram with up 2.976 values values (.e. g. a measurement each 15 min for a whole month. That one doesn't come up in Fyne app.

### Example code:

We, and so you may use 
canvas.NewImageFromReader:  No issue
github.com/ajstarks/svgo/ svg package: May make no difference
browser.OpenReader for benchmark (github.com/pkg/browser): Good results with any standard browser
The refresh mechanims is as simple as here https://developer.fyne.io/started/updating

### Device (please complete the following information):
 - Windows 10
 - **Version:** 
 - Go 1.18.3
 - Fyne v2.2.1

**Comment by Bluebugs** [comment]
Can you share an example of the slow svg?

**Comment by andydotxyz** [comment]
I am a little confused by this:

> The larger, but simple SVG image is a bar diagram with up 2.976 values values (.e. g. a measurement each 15 min for a whole month. That one doesn't come up in Fyne app.

You are displaying an image with 15 min accuracy. So how fast do you really need to refresh it?

If you are looking to have lots of data move and react fast it is more recommended to use the canvas primitives, as they avoid the parse and rasterise steps of SVG loading.

**Comment by EugeneSindambiwe** [comment]
@Andy: I am talking about two cases: In the first one, the SVG image is pretty small and simple. Only that one needs to refresh.  The second test case involves a bigger but also simple SVG image. This does not need  to refresh. It is this one that not even comes up in a Fyne app.
@Andy, @Cedric: Below is a copy of each one. Save as .html and open with a browser. It opens very quickly. Please note that I create them in memory and not read from file. I guess this should not make a difference. 
Thanks
[BIG_SVG.docx](https://github.com/fyne-io/fyne/files/9034529/BIG_SVG.docx)
[SMALL_SVG.docx](https://github.com/fyne-io/fyne/files/9034533/SMALL_SVG.docx)

**Comment by andydotxyz** [comment]
I guess this is the desired output?

![BIG_SVG](https://user-images.githubusercontent.com/294436/177045922-cf65f1aa-d467-40ad-8c1f-139dcb84b738.svg)


I don't know why it takes so long. I guess we would have to analyse the CPU usage through the various elements of the SVG library we use.

**Comment by EugeneSindambiwe** [comment]
Right. Thanks a lot for quick return. I wanted to provide feedback for later improvments instead of passively waiting. Another thought was, since so many browsers are open sourced (code) including in languages with lower overall performance than Go, a look at their implementation of SVG rendering might be helpful. 
From own experience I know how performance analysis and improvement can be time consuming. Unfortunately, for lack of time, I am unable to contribute to the source code analysis as of now. However, I am open-end patient to any updates of yours.

**Comment by andydotxyz** [comment]
> Another thought was, since so many browsers are open sourced (code) including in languages with lower overall performance than Go, a look at their implementation of SVG rendering might be helpful.

Probably not - their licenses typically do not allow code to be grabbed and added to a permissive codebase such as Fyne.
